enhancements/sdk-metrics.md

```diff
@@ -0,0 +1,120 @@
+
+### Goals
+
+The goal of this enhancement is to :
+1.  List the bundle resources on which SDK stamp will be added.
+2. The format of the stamp which will be added.
+
+### Design details
+
+The current bundle image has the following project structure:
+
+```
+ ...
+  bundle.Dockerfile
+  ...
+  $ tree ./deploy/olm-catalog/test-operator/
+  ./deploy/olm-catalog/test-operator/
+  ├── manifests
+  │   ├── example.com_tests_crd.yaml
+  │   └── test-operator.clusterserviceversion.yaml
+  └── metadata
+      └── annotations.yaml
+```
+
+SDK stamps would be present in the following locations:
+
+1. bundle.Dockerfile
+
+The dockerfile will include the sdk stamp in the form of `LABEL` which will follow the following format:
+
+```
+LABEL operators.operatorframework.io.bundle.sdk.v1 = operator_type:sdk_version
+```
+
+2. annotations.yaml
+
+Similar to the format followed in the dockerfile, `annotations.yaml` will have the following label:
+
+```
+operators.operatorframework.io.bundle.sdk.v1: operator_type:sdk_version
+```
+
+For example, in case of memcached-operator, the labels can be written as:
+
+```
+LABEL operators.operatorframework.io.bundle.sdk.v1 = go:v0.17.0
+```
+
+3. Cluster service version
+
+The cluster service version generated by SDK using the `operator-sdk generate csv` command can have a similar stamp included in the metadata. The stamp will be present as an [annotation][kubernetes_annotation]. For example, while generating a new CSV, the following will be added to Object metadata:
+
+```Go
+func newCSV(name, version string) (*olmapiv1alpha1.ClusterServiceVersion, error) {
+    csv := &olmapiv1alpha1.ClusterServiceVersion{
+        ...
+        ObjectMeta: metav1.ObjectMeta{
+			Name:      getCSVName(name, version),
+			Namespace: "placeholder",
+			Annotations: map[string]string{
+				"capabilities": "Basic Install",
+                "csv-builder": "sdk-go-v0.17.0",
+			},
+        }
+        ...
+}
+```
+
+The corresponding CSV will appear to be:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    capabilities: Basic Install
+    csv-builder: "sdk-go-v0.17.0"
+    ...
+```
+
+This can further be read by OLM while recording the metrics on the status of CSVs.
+
+### Limitations
+
+1. The metrics which would be collected through this process will not cover the audience which uses Operator SDK but not OLM to deploy and run their operators.
+2. Currently, `operator-sdk generate csv` officially supports only Go Operators. Ansible and Helm Operators will be fully supported in future. Hence, the collected metrics would majorly cover only Go operators.
```

Comment:
Officially this is true. However the generator will still work (minimally) for these project types, so we can still inject annotations.

Reply:
General bundle question: can Ansible and Helm operator-managed projects be bundled? I know there's a `helm` mediatype, but is that for a set of Helm charts and not necessarily an operator?

If they can be, then we can treat any operator project type managed by the SDK the same, since we can generate a `bundle.Dockerfile` and apply our labels for those types too.

Reply:
> can Ansible and Helm operator-managed projects be bundled?

Yes, it's just that the features of CSV generation that use Go source code obviously won't work for Ansible and Helm operators since those projects don't contain Go source. If we haven't already we should exercise the proposed bundle generation workflow on ansible and helm operators and see what happens and if there are any major gaps.